spec/factories/budgets.rb

```diff
@@ -111,7 +111,10 @@
 
   factory :budget_investment, class: "Budget::Investment" do
     sequence(:title) { |n| "Budget Investment #{n} title" }
-    heading { budget&.headings&.reload&.sample || association(:budget_heading, budget: budget) }
+    heading do
+      budget_headings = budget&.headings&.reload
+      budget_headings&.sample || association(:budget_heading, budget: budget)
+    end
```

Comment:
Not sure about this change :thinking:. Let's talk about it :smile:.

Reply:
I can imagine your concern is that going from one line to four doesn’t really justify the minimal benefit of introducing this rule. With that in mind, I’ll remove the commit that adds the rule and revert these changes 👌

Reply:
One line to four isn't a big deal; the original code could be written in three lines (switching to a `do...end` block) and it'd be fine with me. My concern is more that I find the intermediate variable assignment harder to follow :thinking:.